### ISSUE TITLE
23$: Improve error on invalid JS expression in JSX <script>

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#884]
In the following JSX code, `{` is interpreted as a JS escape. `init();` is not a valid JavaScript expression, so the code has a syntax error:
```javascript
return <script>
  window.onload = () => {
    init();
  };
</script>;
```
We should provide a nice diagnostic, like:
> '{' begins JavaScript code; enclose your <script> in '{\`' and '\`}' instead

See also #883.
